vault/vault.go

```diff
@@ -478,3 +482,10 @@ func (k keyPath) Path() string {
 func (k keyPath) String() string {
 	return fmt.Sprintf("backendPath=%s, backendV2=%t, namespace=%s, secretID=%s", k.backendPath, k.isBackendV2, k.namespace, k.secretID)
 }
+
+func closeIdleConnections(cfg *api.Config) {
+	// close connection in case of error (a fix for go version < 1.12)
+	if tp, ok := cfg.HttpClient.Transport.(*http.Transport); ok {
```

Comment:
Can you add nil checks here for cfg and cfg.HttpClient ?

Reply:
updated